pkg/clusteragent/admission/mutate/autoinstrumentation/env_vars.go

```diff
@@ -0,0 +1,135 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build kubeapiserver
+
+package autoinstrumentation
+
+import (
+	"fmt"
+
+	mutatecommon "github.com/DataDog/datadog-agent/pkg/clusteragent/admission/mutate/common"
+	corev1 "k8s.io/api/core/v1"
+)
+
+const (
+	// Java config
+	javaToolOptionsKey   = "JAVA_TOOL_OPTIONS"
+	javaToolOptionsValue = " -javaagent:/datadog-lib/dd-java-agent.jar -XX:OnError=/datadog-lib/java/continuousprofiler/tmp/dd_crash_uploader.sh -XX:ErrorFile=/datadog-lib/java/continuousprofiler/tmp/hs_err_pid_%p.log"
+
+	// Node config
+	nodeOptionsKey   = "NODE_OPTIONS"
+	nodeOptionsValue = " --require=/datadog-lib/node_modules/dd-trace/init"
+
+	// Python config
+	pythonPathKey   = "PYTHONPATH"
+	pythonPathValue = "/datadog-lib/"
+
+	// Dotnet config
+	dotnetClrEnableProfilingKey   = "CORECLR_ENABLE_PROFILING"
+	dotnetClrEnableProfilingValue = "1"
+
+	dotnetClrProfilerIDKey   = "CORECLR_PROFILER"
+	dotnetClrProfilerIDValue = "{846F5F1C-F9AE-4B07-969E-05C26BC060D8}"
+
+	dotnetClrProfilerPathKey   = "CORECLR_PROFILER_PATH"
+	dotnetClrProfilerPathValue = "/datadog-lib/Datadog.Trace.ClrProfiler.Native.so"
+
+	dotnetTracerHomeKey   = "DD_DOTNET_TRACER_HOME"
+	dotnetTracerHomeValue = "/datadog-lib"
+
+	dotnetTracerLogDirectoryKey   = "DD_TRACE_LOG_DIRECTORY"
+	dotnetTracerLogDirectoryValue = "/datadog-lib/logs"
+
+	dotnetProfilingLdPreloadKey   = "LD_PRELOAD"
+	dotnetProfilingLdPreloadValue = "/datadog-lib/continuousprofiler/Datadog.Linux.ApiWrapper.x64.so"
+
+	// Ruby config
+	rubyOptKey   = "RUBYOPT"
+	rubyOptValue = " -r/datadog-lib/auto_inject"
+
+	// EnvNames
+	instrumentationInstallTypeEnvVarName = "DD_INSTRUMENTATION_INSTALL_TYPE"
+	instrumentationInstallTimeEnvVarName = "DD_INSTRUMENTATION_INSTALL_TIME"
+	instrumentationInstallIDEnvVarName   = "DD_INSTRUMENTATION_INSTALL_ID"
+
+	// Values for Env variable DD_INSTRUMENTATION_INSTALL_TYPE
+	singleStepInstrumentationInstallType   = "k8s_single_step"
+	localLibraryInstrumentationInstallType = "k8s_lib_injection"
+)
+
+var (
+	singleStepInstrumentationInstallTypeEnvVar = corev1.EnvVar{
+		Name:  instrumentationInstallTypeEnvVarName,
+		Value: singleStepInstrumentationInstallType,
+	}
+
+	localLibraryInstrumentationInstallTypeEnvVar = corev1.EnvVar{
+		Name:  instrumentationInstallTypeEnvVarName,
+		Value: localLibraryInstrumentationInstallType,
+	}
+)
+
+type envVar struct {
```

Comment:
FYI `envVar` and all of the `envValFunc` definitions below are moved over from `auto_instrumentation.go`. 

The only addition here is that `envVar` is now a `containerMutator`